scrapy/http/request/__init__.py

```diff
@@ -36,6 +36,7 @@ def __init__(self, url, callback=None, method='GET', headers=None, body=None,
         self.dont_filter = dont_filter
 
         self._meta = dict(meta) if meta else None
+        self.flags = None if flags is None else list(flags)
```

Comment:
Is this to address @kmike 's https://github.com/scrapy/scrapy/pull/2082#issuecomment-231215967 ?
I haven't tested but does "_users should be able to do flags.append without checking for None_ work"?

If `None` is to be kept, what about `self.flags = list(flags) if flags else None`?